File: Lab 3/C#/StringNumber.cs
using System;

namespace Artem
{
    public class StringNumber
    {
        private string value;

          ////////////////////
         //  Конструктори  //
        ////////////////////

        public StringNumber()
        {
            value = string.Empty;
        }
        
        public StringNumber(string val)
        {
            value = val;
        }

        public StringNumber(StringNumber other)
        {
            value = other.value;
        }

          //////////////
         //  Методи  //
        //////////////

        public int Length()
        {
            return value.Length;
        }       
        public string GetValue()
        {
            return value;
        }

          ////////////////////////
         //  Перенавантаження  //
        ////////////////////////

        public static StringNumber operator +(StringNumber sn1, StringNumber sn2)
        {
            return new StringNumber(sn1.value + sn2.value);
        }
        public static StringNumber operator -(StringNumber sn, char number)
        {
            string newValue = string.Empty;

            foreach (char c in sn.value)
            {
                if (c != number)
                {
                    newValue += c;
                }
            }

            return new StringNumber(newValue);
        }
    }
}

File: Lab 3/C++/StringNumber.cpp
#include "StringNumber.h"
using namespace Artem;

////////////////////
//  Конструктори  //
////////////////////

StringNumber::StringNumber()//за умовчуванням створює порожній рядок  
{
    value = "";
}
StringNumber::StringNumber(const string& val)//числа зберігаються в strin value
{
    value = val;
}
StringNumber::StringNumber(const StringNumber& other)//копіює значення 
{
    value = other.value;
}

//////////////
//  Методи  //
//////////////

size_t StringNumber::length() const 
{
    return value.length();
}
string StringNumber::getValue() const//отримання значення рядка
{
    return value;
}

////////////////////////
//  Перенавантаження  //
////////////////////////

StringNumber StringNumber::operator+(const StringNumber& other) const 
{
    return StringNumber(value + other.value);
}
StringNumber StringNumber::operator-(char number) const 
{
    string newValue;
    for (size_t i = 0; i < value.length(); ++i) 
        {
        if (value[i] != number) 
        {
            newValue += value[i];//записуємо символи, які не є рівними number
        }
    }
    return StringNumber(newValue);
}
